ingestion/src/metadata/ingestion/source/database/common_db_source.py

```diff
@@ -0,0 +1,218 @@
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+from metadata.ingestion.ometa.utils import ometa_logger
+from metadata.ingestion.source.database.database_source import DatabaseSourceService
+from metadata.ingestion.source.database.dbt_souce import DBTSource
+from metadata.ingestion.source.database.sql_column_handler import SqlColumnHandler
+from metadata.utils.connections import (
+    create_and_bind_session,
+    get_connection,
+    test_connection,
+)
+
+logger = ometa_logger()
+
+
+@dataclass
+class SQLSourceStatus(SourceStatus):
+    """
+    Reports the source status after ingestion
+    """
+
+    success: List[str] = field(default_factory=list)
+    failures: List[str] = field(default_factory=list)
+    warnings: List[str] = field(default_factory=list)
+    filtered: List[str] = field(default_factory=list)
+
+    def scanned(self, record: str) -> None:
+        self.success.append(record)
+        logger.info(f"Table Scanned: {record}")
+
+    def filter(self, record: str, err: str) -> None:
+        self.filtered.append(record)
+        logger.warning(f"Filtered Table {record} due to {err}")
+
+
+class CommonDbSourceService(DBTSource, SqlColumnHandler, DatabaseSourceService):
+    def __init__(
+        self,
+        config: WorkflowSource,
+        metadata_config: OpenMetadataConnection,
+    ):
+        self.config = config
+        self.source_config: DatabaseServiceMetadataPipeline = (
+            self.config.sourceConfig.config
+        )
+        # It will be one of the Unions. We don't know the specific type here.
+        self.service_connection = self.config.serviceConnection.__root__.config
+        self.status = SQLSourceStatus()
+
+        self.metadata_config = metadata_config
+        self.metadata = OpenMetadata(metadata_config)
+
+        self.service = self.metadata.get_service_or_create(
+            entity=DatabaseService, config=config
+        )
+        self.engine: Engine = get_connection(self.service_connection)
+        self.test_connection()
+
+        self._session = None  # We will instantiate this just if needed
+        self._connection = None  # Lazy init as well
+        self.data_profiler = None
+        self.data_models = {}
+        self.table_constraints = None
+        self.database_source_state = set()
+        self.profile_date = datetime.now()
+        super().__init__()
+
+    def prepare(self):
+        self._parse_data_model()
+
+    @classmethod
+    def create(cls, config_dict: dict, metadata_config: OpenMetadataConnection):
+        pass
+
+    def standardize_schema_table_names(
+        self, schema: str, table: str
+    ) -> Tuple[str, str]:
+        return schema, table
+
+    def fetch_sample_data(self, schema: str, table: str) -> Optional[TableData]:
+        """
+        Get some sample data from the source to be added
+        to the Table Entities
+        """
+        try:
+            query = self.source_config.sampleDataQuery.format(schema, table)
```

Comment:
Note to the future -> the sampleDataQuery is actually something that could become a custom SQA function similar to the ones we have for the ORM profiler. It would make things easier and that way we won't force users to pass that